Chromium integration

jaragunde opened the pull request:
These patches add the required files to integrate Chromium with the GENIVI HMI and append some rules to the chromium recipe to workaround a couple of issues with the window aspect ratio and the icon.

Depends on the [meta-browser layer](https://github.com/OSSystems/meta-browser), which needs to be added as a submodule to [genivi-dev-platform](https://github.com/GENIVI/genivi-dev-platform) on a future PR. EDIT: PR sent: https://github.com/GENIVI/genivi-dev-platform/pull/57

jaragunde committed:
chromium: Integration with GENIVI HMI

Add the .desktop file to integrate Chromium with the GENIVI HMI.
Also make sure icons of all sizes are installed, so the 128px icon
can be used for the HMI menu.

Built upon the meta-browser layer from:
  https://github.com/OSSystems/meta-browser

[GDP-477] Integrate Chromium on a Wayland/Ozone stack

Signed-off-by: Jacobo Aragunde Pérez <jaragunde@igalia.com>

jaragunde committed:
chromium: Force browser window size

Applications have no knowledge of expected surface size and the GDP HMI
stretches them to fill all the available space. According to
LayerManagerControl, the destination region for applications in the HMI
has a 1728x1080 size, so forcing the browser window to have that size
is a workaround to ensure the browser has the proper aspect ratio.

[GDP-477] Integrate Chromium on a Wayland/Ozone stack
[LM-2] Applications under ILM need to know expected surface size

Signed-off-by: Jacobo Aragunde Pérez <jaragunde@igalia.com>

chbae commented on meta-genivi-dev/recipes-extended/chromium/chromium-wayland_%.bbappend:
```diff
@@ -0,0 +1,15 @@
+do_install_append() {
+
+    install -d ${D}/usr/share/applications/
+    install -m 0444 ${S}/../chromium.app ${D}/usr/share/applications/
```
${S}../ -> ${WORKDIR}
/usr/share -> ${datadir}

chbae commented on meta-genivi-dev/recipes-extended/chromium/chromium-wayland_%.bbappend:
```diff
@@ -0,0 +1,15 @@
+    install -m 0444 ${S}/../chromium.app ${D}/usr/share/applications/
+
+    install -d ${D}${libdir}/systemd/user
+    install -m 0444 ${S}/../chromium.service ${D}${libdir}/systemd/user/
```
${S}../ -> ${WORKDIR}

chbae commented on meta-genivi-dev/recipes-extended/chromium/chromium-wayland_%.bbappend:
```diff
@@ -5,6 +5,9 @@ SRC_URI += "\
 "
 
+# FIXME: workaround for https://at.projects.genivi.org/jira/browse/LM-2
+CHROMIUM_EXTRA_ARGS += " --window-size=1728,1080"
```
CHROMIUM_EXTRA_ARGS_append = " --window-size=1728,1080"

jaragunde commented:
Thanks for the feedback, sending updated versions of the patches.

chbae commented on meta-genivi-dev/recipes-extended/chromium/chromium-wayland_%.bbappend:
```diff
@@ -0,0 +1,15 @@
+FILESEXTRAPATHS_append := "${THISDIR}/${PN}"
```
FILESEXTRAPATHS_prepend := "${THISDIR}/${PN}:"

chbae commented on meta-genivi-dev/recipes-extended/chromium/chromium-wayland_%.bbappend:
```diff
@@ -0,0 +1,15 @@
+"
+
+do_install_append() {
+
```
please remove this line.

chbae commented:
Sorry I add more comments. I missed before.

jaragunde commented:
@chbae Thanks for the review! I've updated the patches.

zeenix reviewed (changes requested):
1. Please keep shortlog as short as possible.

2. Could you add to details a bit. E.g what you are patching and why. That'd help keeping all info while stilling keeping short log short. :)

 Please read https://at.projects.genivi.org/wiki/display/PROJ/How+to+contribute+to+GENIVI and check all your patches for requirements of Patch process (especially 4).

zeenix reviewed (changes requested):
Shortlog too long here too and 'f' in "force" needs to be capitalized (I'm a big nitpicker I know). Good job with providing details in description.

BTW, you are missing Jira ticket reference in your commit messages.

zeenix commented on meta-genivi-dev/recipes-extended/chromium/chromium-wayland_%.bbappend:
```diff
@@ -5,6 +5,9 @@ SRC_URI += "\
 
+# FIXME: workaround for https://at.projects.genivi.org/jira/browse/LM-2
+CHROMIUM_EXTRA_ARGS_append = " --window-size=1728,1080"
+
```
Good, we have a workaroud but I understand you couldn't build for Silk yet so i guess it's not be tested on more than 1 target and monitor?

jaragunde replied:
What options do we have to try a multi-monitor setup, only R-Car boards? I definitely want to try this out. Also, how does the HMI behave in that case?

jaragunde replied:
Oh, I realize I've misunderstood your comment. The workaround should work on a 720p screen because that resolution has the same aspect ratio, and the HMI would downscale the browser surface. I reckon it's a waste, though. It wouldn't work with other resolutions, but those are generally not well supported by embedded hardware so I guess they aren't very interesting.

jaragunde commented:
Pushed new version of the patches with reworded commit messages. I hope they fit better now.

zeenix reviewed (commented):
Not sure why this is a separate change? Shouldn't it be squashed with previous one?

jaragunde commented:
> Not sure why this is a separate change? Shouldn't it be squashed with previous one?

Sorry @zeenix, I think GH is letting some context get lost. I don't know which specific change you are talking about :)

If you prefer, we could squash everything together. The three commits are actually a result of the process I followed: wrote the basics, then fixed individual problems as they were detected.

zeenix commented:
@jaragunde Not at all, I think you just missed the "View changes" button on the right-top of my comment. It's the first two commits, since the second one is just fixing the files that are added in first one. The 3rd patch is a separate change and hence should be left as is.

jaragunde commented:
New patches sent:
* Squash two patches into one (basic setup + 128px icons).
* Address one small change requested by @chbae that I had left behind.

zeenix commented:
Let's merge this immediately after https://github.com/GENIVI/genivi-dev-platform/pull/59

jaragunde committed:
chromium: Add to genivi-dev-platform image

Add the chromium-wayland browser to the GDP default build.

[GDP-477] Integrate Chromium on a Wayland/Ozone stack

Signed-off-by: Jacobo Aragunde Pérez <jaragunde@igalia.com>

zeenix commented on meta-genivi-dev/recipes-dev-platform/images/genivi-dev-platform.bb:
```diff
@@ -32,6 +32,7 @@ IMAGE_INSTALL_append = " \
     packagegroup-gdp-qt5 \
     packagegroup-gdp-rvi \
     packagegroup-gdp-dev \
+    chromium-wayland \
```
Without this change, wouldn't the first change be broken? A patch shouldn't depend on a following patch in the tree/series but only the other way around. Maybe this patch should just come first?

jaragunde replied:
I like this order better because it guarantees that Chromium is built with the GDP specific bits that we added in the previous patches. Previous patches alone don't break, they just allow to build Chromium optionally. Anyway, it doesn't make a big difference, the build shouldn't break in any order.

zeenix replied:
I mean if you don't have Chromium included in the image, wouldn't that mean HMI will have a button to launch an app that doesn't exist? I might be misunderstanding something here though.

chbae replied:
Why don't you add "chromium-wayland" in packagegroup-gdp-browser.bb?

jaragunde replied:
@zeenix The HMI button is added by the chromium-wayland recipe and it will only run if you add Chromium to the image, so that won't be a problem. If chromium is not in the image, the button will not be added.

@chbae: oh, that looks cleaner, indeed. Let me change it.

zeenix reviewed (changes requested):
Just one comment

chbae commented:
Just wondering.
- This chromium browser supports all devices like raspberrypi, minnowboard and othres?

jaragunde commented:
Last patch moved `chromium-wayland` to `packagegroup-gdp-browser`.

> This chromium browser supports all devices like raspberrypi, minnowboard and othres?

* Minnowboard works, it's what we are currently working on.
* Raspberry Pi is not building due to [this issue](http://lists.openembedded.org/pipermail/openembedded-devel/2017-February/110973.html), a person said he had patches for this issue in that mail thread, let me ping him again.
* R-Car Gen 2 should work, we did Porter builds some time ago but we cannot test that at the moment.

zeenix commented:
@jaragunde Oh so this breaks build for RPI? We can't merge this yet then.

In future, if there are known breakages, I'd like that specified in commit logs and/or the PR summary.

gunnarx commented:
@chbae, you are the main R-Pi expert.  Could you help out on the R-Pi trouble?

jaragunde committed:
chromium: Fix compilation for Raspberry Pi

The Raspberry Pi is identified as an ARMv7VE board and Yocto sets the
corresponding optimization flags. Unfortunately, Chromium sets its own
flags too, and assumes that any ARMv7 compilation is ARMv7A, which sets
a set of flags that conflict Yocto's.

Besides, meta-browser recipe for Chromium also expects the ARMv7A
architecture, and refuses to build if it's not set.

This patch adds certain overrides for ARMv7VE architectures, which let
the recipe behave like in the case of an ARMv7A build:

* Add armv7ve as a compatible architecture.
* Use armv7a FP tune flags.
* Add a set of source files that come from the armv7a build.
* Disable the incompatible armv7ve optimization flag.

[GDP-477] Integrate Chromium on a Wayland/Ozone stack

Signed-off-by: Jacobo Aragunde Pérez <jaragunde@igalia.com>

gunnarx commented:
Build checks:  Minnowboard link is no longer available because I removed the pipeline.  Talked before to @zeenix about limiting the number of pipelines somewhat, and Minnowboard compilation should be quite similar to the QEMU build we do for Intel architecture, so now Dragonboard (Arm-64 architecture) is included instead.   (Note that for G-D-P PRs, **all** targets are built, so there's still another check that would catch any problem once the submodule bump is proposed, before it reaches master).

gunnarx commented:
When this PR is rebased we should get new checks probably.

jaragunde commented:
Rebased and pushed the branch again; added a set of workarounds to be able to compile Chromium on the Raspberry Pi. The build now completes successfully but, unfortunately, the resulting browser is not running yet.

jaragunde commented:
According to the build results for Porter, it seems we now need a similar workaround to the one we did for the Raspberry Pi.

Error message is:
```
error: switch -mcpu=cortex-a15 conflicts with -march=armv7-a switch [-Werror]

gunnarx commented:
Pushed this to [a branch](https://github.com/gunnarx/genivi-dev-platform/tree/chromium) for easier handling and started separate CI pipeline group to easier track/update/finish this.  (Actually I ought to move it to a branch on GENIVI repo so that others can also push to it...)

DunderRoffe commented:
> Pushed this to a branch<https://github.com/gunnarx/genivi-dev-platform/tree/chromium> for easier handling and started separate CI pipeline group to easier track/update/finish this. (Actually I ought to move it to a branch on GENIVI repo so that others can also push to it...)


Nice. Collaborating on the same branch is a hassle so I suggest we pull from your fork and push to our own.


—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/GENIVI/meta-genivi-dev/pull/53#issuecomment-296837973>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AFrh7b0OAuWN8P-6Qspn7eAPSg_6JyWHks5rzR_EgaJpZM4LoaId>.

gunnarx commented:
> Nice. Collaborating on the same branch is a hassle so I suggest we pull from your fork and push to our own.
OK well then be aware I'm still rebasing, amending and and force-pushing a lot to my own fork...  There's also the issue of main project <-> submodule...

jaragunde commented:
Hi @gunnarx, I'm currently working on:
* rebase on top of latest meta-genivi-dev master
* replace the .app file with a .desktop per https://github.com/GENIVI/meta-genivi-dev/commit/7a4c56cd9fef2f80e67e5c9a380692681c7bb95c
* add an extra commit to support R-Car gen.3 boards

I was planning to push changes here. To prevent stepping into each other's toes, what are you currently doing on top if this branch? :)

DunderRoffe commented:
I found that chromium already installs a .desktop file on the system so it might not be necessary to add it @jaragunde. However, while looking at the .app and .service file they refer to different binaries what is the difference between them?

jaragunde commented:
@DunderRoffe the original .app I wrote and the .desktop file that chromium generates are different in:
* .app uses a bigger resolution icon to match the icon sizes in the HMI
* .app includes a `Unit` entry to match the corresponding systemd service.
* .app points to the `/usr/bin/chromium/chrome` binary, and .desktop to `/usr/bin/google-chrome`. The latter is actually a shell script that wraps the former. When I created the .app file, I found that the HMI controller needed the name of the actual binary to properly identify the browser surfaces.

jaragunde committed:
chromium: Backport patch to fix ILL_ILLOPN error

Problem appears when running kernel 4.5 or higher. Backport the
upstream patch that fixes the issue to our Chromium version.

Technical details can be found in the patch itself or in
https://codereview.chromium.org/2490893002

[GDP-477] Integrate Chromium on a Wayland/Ozone stack

Signed-off-by: Jacobo Aragunde Pérez <jaragunde@igalia.com>

DunderRoffe commented:
Ok thx for explaining :)

When I try to run the images built by the CI system on the Minnowboard or RaspberryPi3 I get the following error:
```
[1354:1354:0426/080855:ERROR:desktop_window_tree_host_ozone.cc(579)] Not implemented reached in virtual void views::DesktopWindowTreeHostOzone::SetWindowIcons(const gfx::ImageSkia&, const gfx::ImageSkia&)
[1354:1354:0426/080855:ERROR:ozone_webui.cc(61)] Not implemented reached in virtual void views::OzoneWebUI::GetDefaultFontDescription(std::__cxx11::string*, int*, int*, gfx::Font::Weight*, gfx::FontRenderParams*) const
[1354:1354:0426/080855:ERROR:ozone_webui.cc(51)] Not implemented reached in virtual gfx::FontRenderParams views::OzoneWebUI::GetDefaultFontRenderParams() const
[1354:1354:0426/080855:ERROR:ozone_webui.cc(51)] Not implemented reached in virtual gfx::FontRenderParams views::OzoneWebUI::GetDefaultFontRenderParams() const
[1354:1354:0426/080855:ERROR:ozone_webui.cc(51)] Not implemented reached in virtual gfx::FontRenderParams views::OzoneWebUI::GetDefaultFontRenderParams() const
[1354:1354:0426/080855:ERROR:desktop_window_tree_host_ozone.cc(579)] Not implemented reached in virtual void views::DesktopWindowTreeHostOzone::SetWindowIcons(const gfx::ImageSkia&, const gfx::ImageSkia&)
[1354:1354:0426/080855:ERROR:ozone_webui.cc(51)] Not implemented reached in virtual gfx::FontRenderParams views::OzoneWebUI::GetDefaultFontRenderParams() const
[1354:1354:0426/080855:ERROR:ozone_webui.cc(51)] Not implemented reached in virtual gfx::FontRenderParams views::OzoneWebUI::GetDefaultFontRenderParams() const
[1385:1385:0426/080855:ERROR:sandbox_linux.cc(343)] InitializeSandbox() called with multiple threads in process gpu-process. 
Received signal 4 ILL_ILLOPN 5634e72e45e0
```

Furthermore, the UI starts up on the Minnowboard but not on the RPi3. However, even if the UI starts it only displays "Aw, Snap! Something went wrong while displaying this webpage". Trying to open another webpage results in the same output. Do you have any ideas about this? @jaragunde

jaragunde commented:
Hi @DunderRoffe, I'm working on a rebase of this branch, you can check the ongoing process [here](https://github.com/jaragunde/meta-genivi-dev/tree/chromium-integration-rebase). Regarding the problems you mention:
* Chromium on RPi is still not running. This is a known issue related with graphic drivers we need to look into. There is an [interesting discussion](https://lists.yoctoproject.org/pipermail/yocto/2017-April/035642.html) in Yocto ML regarding the topic.
* "Aw, Snap!" on Minnowboard: kernel upgrade in GDP has caused this issue to surface. It's fixed with a [backported Chromium patch](https://github.com/jaragunde/meta-genivi-dev/commit/20bb10e16541fbd44727f5f4a7e392a113b1a608) available in my rebase branch.

My intention is to push that rebase branch here to continue the review process.

DunderRoffe commented:
This might not be news to you but I thought I share my findings after building a new image using the branch you referenced.

When run from HMI it loads but does not accept any input. However, it does work when launched from the command line need to investigate this further...

Anyway when running from command line it works but I still get these printouts from starting the app:
```
[732:732:0426/152323:ERROR:desktop_window_tree_host_ozone.cc(579)] Not implemented reached in virtual void views::DesktopWindowTreeHostOzone::SetWindowIcons(const gfx::ImageSkia&, const gfx::ImageSkia&)
[732:732:0426/152323:ERROR:ozone_webui.cc(61)] Not implemented reached in virtual void views::OzoneWebUI::GetDefaultFontDescription(std::__cxx11::string*, int*, int*, gfx::Font::Weight*, gfx::FontRenderParams*) const
[732:732:0426/152323:ERROR:ozone_webui.cc(51)] Not implemented reached in virtual gfx::FontRenderParams views::OzoneWebUI::GetDefaultFontRenderParams() const
[732:732:0426/152323:ERROR:ozone_webui.cc(51)] Not implemented reached in virtual gfx::FontRenderParams views::OzoneWebUI::GetDefaultFontRenderParams() const
[732:732:0426/152323:ERROR:ozone_webui.cc(51)] Not implemented reached in virtual gfx::FontRenderParams views::OzoneWebUI::GetDefaultFontRenderParams() const
[732:732:0426/152323:ERROR:desktop_window_tree_host_ozone.cc(579)] Not implemented reached in virtual void views::DesktopWindowTreeHostOzone::SetWindowIcons(const gfx::ImageSkia&, const gfx::ImageSkia&)
[732:732:0426/152323:ERROR:ozone_webui.cc(51)] Not implemented reached in virtual gfx::FontRenderParams views::OzoneWebUI::GetDefaultFontRenderParams() const
[732:732:0426/152323:ERROR:ozone_webui.cc(51)] Not implemented reached in virtual gfx::FontRenderParams views::OzoneWebUI::GetDefaultFontRenderParams() const
[763:763:0426/152323:ERROR:sandbox_linux.cc(343)] InitializeSandbox() called with multiple threads in process gpu-process. 
```
As well as these when loading a page:
```
[732:733:0426/152340:ERROR:ozone_webui.cc(51)] Not implemented reached in virtual gfx::FontRenderParams views::OzoneWebUI::GetDefaultFontRenderParams() const
[732:733:0426/152340:ERROR:ozone_webui.cc(51)] Not implemented reached in virtual gfx::FontRenderParams views::OzoneWebUI::GetDefaultFontRenderParams() const
[732:733:0426/152340:ERROR:ozone_webui.cc(51)] Not implemented reached in virtual gfx::FontRenderParams views::OzoneWebUI::GetDefaultFontRenderParams() const
[732:733:0426/152340:ERROR:ozone_webui.cc(51)] Not implemented reached in virtual gfx::FontRenderParams views::OzoneWebUI::GetDefaultFontRenderParams() const
[732:733:0426/152340:ERROR:ozone_webui.cc(51)] Not implemented reached in virtual gfx::FontRenderParams views::OzoneWebUI::GetDefaultFontRenderParams() const
[732:733:0426/152340:ERROR:ozone_webui.cc(51)] Not implemented reached in virtual gfx::FontRenderParams views::OzoneWebUI::GetDefaultFontRenderParams() const
```

jaragunde commented:
Hi @DunderRoffe,

> When run from HMI it loads but does not accept any input.

This is happening since the change in how hmi launches the applications. The problem is the `Exec` field in the .desktop file, it points to a shell script that wraps the actual chromium binary. In this situation, the HMI seems to be unable to identify that the application surface has popped up and doesn't assign focus.

Quick test: ssh into the board while chromium is running:
```
root@intel-corei7-64:~# LayerManagerControl get input focus
surface 7001:  <- this is chromium
surface 8413: pointer keyboard touch
surface 0: 
```
Now modify `/usr/share/applications/google-chrome.desktop`, set `Exec=/usr/bin/chromium/chrome`, then reboot, launch chromium again. Now the browser will issue a modal window and refuse to run, but will input properly assigned:
```
# LayerManagerControl get input focus
surface 7001: pointer keyboard touch
surface 8436: 
surface 0: 
```

Finally, if I set the required command-line flags to chromium in the .desktop file, it isn't properly read by the HMI and the chromium launcher doesn't appear: `Exec=/usr/bin/chromium/chrome --use-gl=egl --window-size=1728,1080 --user-data-dir=/home/root/.chromium/`. I tried to use double quotes to escape the parameters as [in the spec](https://specifications.freedesktop.org/desktop-entry-spec/latest/ar01s06.html), in that case the launcher appears but has no effect.

How should we proceed regarding these issues? Report to HMIL on JIRA?

DunderRoffe commented:
Due to the time constraints I feel that it is better for me to fix this issue. Also I implemented that feature so if it is broken I might know how to fix it :)

Will try to fix it today

jaragunde commented:
@DunderRoffe Thanks, it's really appreciated. Please let me know if I can help you somehow.

chbae commented:
Oops. I missed this PR. Sorry I will follow up this soon.

jaragunde commented:
Thanks for stopping by @chbae, there actually haven't been new commits to this PR, I'm holding them back in [this branch](https://github.com/jaragunde/meta-genivi-dev/tree/chromium-integration-rebase), although if we consider that's the HMI who can't handle some cases in .desktop files properly, then the rebase of chromium patches is completed and they can already be pushed here.

chbae commented:
@jaragunde Ok. Now I'm building based on latest commit in master branch.

chbae commented:
@jaragunde 
This is related with gpu  as you known. I'm not a export of graphics. Sorry.
Maybe we should find the problem in drm (mesa) and gpu part of chromium.
I know that igalia already done chromium on raspberrypi.

DunderRoffe commented:
Hi,
the latest version of the HMI contains a fix that allows apps to have child processes that show the UI.
I have tested it with the Chromium build and it seems to work. Could you try if this solves the problem @jaragunde? :)

gunnarx commented:
So everyone knows, we're tracking and building some type of release image for this feature using a the feature branch named **chromium** in [meta-genivi-dev](https://github.com/genivi/meta-genivi-dev/tree/chromium) and [genivi-dev-platform](https://github.com/genivi/genivi-dev-platform/tree/chromium), rebasing from time to time.  

Since @jaragunde is also working on a separate branch(es), I might ask if this PR has any value now.
Close it?  I'd say if we need more general feature discussion and planning, move that back to JIRA.  Opinions?

gunnarx commented:
(Please continue short-term technical discussion as needed though @DunderRoffe)

jaragunde commented:
@gunnarx, I think it's worth keeping this PR open and always pointing to the latest version of the integration patches. Tomorrow I will backup the commits in the current `chromium-integration` branch, because it is useful to do GDP11 builds with Chromium, then update this PR with everything from the `chromium-integration-rebase` branch.

jaragunde commented:
Pushed a new set of patches for Chromium integration. What's new:
* Rebase on top of latest master.
* Use a .desktop file for the HMI to launch it.
* Backport patch that fixes a crash with kernel >=4.5.

Regarding hardware support:
* Minnowboard should work.
* RPi compiling, not running. Same issues with graphics stack mentioned earlier.
* Patch for R-Car Gen 3 boards hasn't been included in the PR. The reason is it breaks the Minnowboard build. We need to split it in two patches and commit one of them to Renesas BSP. Already in touch with @slawr on how to proceed with that.
  * Besides, Chromium may segfault on R-Car Gen 3 boards for yet unknown reasons. Other applications crash similarly, we still have to investigate that.

More info:
* Patches for R-Car Gen 3 are still available [in this branch](https://github.com/jaragunde/meta-genivi-dev/tree/chromium-integration-rebase). They can be cherry-picked if somebody wants to use them.
* Older patches in this PR have been saved to a [new branch based on GDP11](https://github.com/jaragunde/meta-genivi-dev/tree/chromium-integration-gdp11), so anyone could build Chromium on the previous release if they wanted.

DunderRoffe commented:
Hi @jaragunde
How do you want to proceed with this PR? I suggest that you move all relevant changes here and then we can continue from that state. 

Furthermore, do you have any suggestion on how to proceed with making it work for the RPi?

jaragunde commented:
Hi @DunderRoffe, the last update I did to this PR contains the latest changes. They should be enough to support the Minnowboard.

I'm taking this week off, when I'm back next week I will take steps to contribute patches to the relevant BSPs for R-Car Gen.3 support. Finally, regarding the Pi, we will also devote some time to that, probably enabling software rendering paths on that platform to work-around the problem.

gunnarx commented:
Summing up the status.

We are currently collecting this work on a feature-branch since that is a bit more flexible in CI etc. than to use the PR.  There is a **chromium** branch of g-d-p and corresponding one in m-g-d.

I based it on the branch @jaragunde  provided shortly before the AMM:
https://github.com/jaragunde/meta-genivi-dev/tree/chromium-integration-rebase

On top of your branch the following changes were added to clean up the R-Car M3 support, according to what we discussed the last 2-3 weeks.

34fb536  chromium: more general machine override for R-car
0f98878  chromium: gcc fix #66758 to meta-linaro-gdp sublayer
For the meta-linaro sublayer change, see corresponding bblayers.conf in [genivi-dev-platform:ff7b12f8](https://github.com/GENIVI/genivi-dev-platform/commit/ff7b12f8eca76e67bf22b230c465ba344e8f971e#diff-a5ea6d2294019bb42d2a56a635764b2a)

However one commit from your branch is not included. In the last build I did I found this one to break something so I removed it again.  The intention is just to tag the current state with minnow and M3 support.  

We can always try the Salvator patch again, maybe it is fine.   In any case I want to add Salvator right after tagging current state.

-> NOT INCLUDED YET:
fffb16c    chromium: support Renesas Salvator-X
https://github.com/jaragunde/meta-genivi-dev/commit/fffb16ca5f87e2ffb0f68f73a5b64369c341b302


Note we are also building in CI so you can see that it works there:
https://go.genivi.org/go/pipelines -- Look for the separate pipeline group named Chromium.


**Multi-seat**

We agreed today that we should create a separate pull request for multi-seat to get it into the pipe...

This PR was long-winded and I feel it is time we try to close up the long standing PR, and move to new more targeted and focused PRs for particular issues from now on.  Let's consider thisPR closed as soon as the minnow and M3 status is fully stable.

gunnarx commented:
Doing the same comparison of our feature-branch **chromium** against the current PR, instead of the  [chromium-integration-rebase](https://github.com/jaragunde/meta-genivi-dev/tree/chromium-integration-rebase) branch:

NOT taken from PR into chromium branch (yet):
228f134   chromium-wayland: Define more boards as compatible
It looks like it replaces: fffb16c chromium: support Renesas Salvator-X

But what about this commit, @jaragunde ?
b64e53d  chromium: Backport patch to fix ILL_ILLOPN error

It's on your "rebase" branch but not in the PR, as far as I can tell.

gunnarx commented:
Please @jaragunde, I need to know if b64e53d is deliberately omitted or not.

jaragunde commented:
@gunnarx that commit was pushed to this PR in the last update (4th May), hash here is 4665bb2 as I rebased the branch on top of master before updating the PR.

jaragunde commented:
@gunnarx My colleague Henrique is currently testing Salvator-X and Minnowboard builds. He has written an additional patch to remove some stuff from the Minnowboard build and set it only for R-Car Gen 3 boards. Once builds finish later today, we will open a PR on your chromium branch.

gunnarx commented:
> @gunnarx that commit was pushed to this PR in the last update (4th May), hash here is 4665bb2 as I rebased the branch on top of master before updating the PR.

OK thanks. I can't understand why I did not see it in the PR.  In any case it was always included in the chromium branch so all is good.

gunnarx commented:
Since we're developing this on a feature branch, let's close this PR?  It just feels like overhead to keep it updated and the communication is better done on the relevant JIRA issues instead.   New targeted PRs can of course be opened, with the feature branch as target.  (It will be on genivi-dev-platform since meta-genivi-dev is being included there)
